### PR TITLE
triplestore class which does add, delete/update and get mappings/triples

### PR DESCRIPTION
**Pull request by Treesarj**

### CODE REVIEW

File: oteapi/triplestore/triplestore.py
Comment (kriwiik) [inline oteapi/triplestore/triplestore.py:100]: I believe this line can be removed.

File: oteapi/triplestore/triplestore.py
Comment (CasperWA) [inline oteapi/triplestore/triplestore.py:113]:
```suggestion
    def update(self, sparql_query: str) -> None:
```
I think `update` is an appropriate name here - isn't this really an overwrite? Or does it truly delete some triples as well completely?

File: requirements.txt
Comment (CasperWA) [inline requirements.txt:1]:
```suggestion
agraph-python>=101.0.7,<102
```

**Review by CasperWA — DISMISSED** [review]

File: oteapi/README.md
Comment (kriwiik) [reply oteapi/README.md:8]: I think triplestore and triple store (no hyphen) are the most commonly used spellings

Comment (CasperWA) [reply oteapi/README.md:8]: https://en.wikipedia.org/wiki/Triplestore Agreed

File: oteapi/models/triplestoreconfig.py
Comment (CasperWA) [reply oteapi/models/triplestoreconfig.py:16]: The `TripleStoreConfig` is dissimilar to all the other `*Config` classes, which represent general strategy configurations.

Rather, `TripleStoreConfig` is similar to the `DataCacheConfig`, upon which it has also been modellen (I believe) :)

File: oteapi/triplestore/triplestore.py
Comment (Treesarj) [reply oteapi/triplestore/triplestore.py]: It will return an empty list

**Comment by jesper-friis** [comment]
Looks very nice. A unit test would be nice to have, but that could also be added in a later PR.

**Comment by jesper-friis** [comment]
Looks very nice. A unit test would be nice to have, but that could also be added in a later PR.

**Comment by Treesarj** [comment]
> Looks very nice. A unit test would be nice to have, but that could also be added in a later PR.

I checked to add a test but it requires a running instance of AllegroGraph. I couldn't find a away ti mock it

**Review by kriwiik — APPROVED** [review]

**Comment by Treesarj** [comment]
dismissing @CasperWA 's review as it's blocking the merge! as discussed I have added the requested changes, but some got outdated due to some line changes

**Pull request merged**
